pkg/aws/validations_test.go

```diff
@@ -0,0 +1,251 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/smithy-go"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/submariner-io/cloud-prepare/pkg/api"
+)
+
+var _ = Describe("AWS Peering", func() {
+	Context("Test Validate Peering Prerequisites", testValidatePeeringPrerequisites)
+	Context("Test Check VPC Overlap", testCheckVpcOverlap)
+	Context("Test Determine Permission Error", testDeterminePermissionError)
+})
+
+func testDeterminePermissionError() {
+	var _ = Describe("Validate error input", func() {
+		When("Args are wrong", func() {
+			It("err is nil", func() {
+				result := determinePermissionError(nil, "")
+				Expect(result).To(BeNil())
+			})
+		})
+		When("It's an AWS error", func() {
+			It("is DryRunOperation error", func() {
+				err := smithy.GenericAPIError{
+					Code:    "DryRunOperation",
+					Message: "DryRunOperation",
+					Fault:   1,
+				}
+				operation := "check"
+				result := determinePermissionError(&err, operation)
+				Expect(result).Should(BeNil())
+			})
+			It("is UnauthorizedOperation error", func() {
+				err := smithy.GenericAPIError{
+					Code:    "UnauthorizedOperation",
+					Message: "UnauthorizedOperation",
+					Fault:   1,
+				}
+				operation := "check"
+				result := determinePermissionError(&err, operation)
+				Expect(result).Should(
+					MatchError(
+						MatchRegexp("no permission to " + operation),
+					),
+				)
+			})
+		})
+		When("It's not an AWS error", func() {
+			It("is a Generic Error", func() {
```

Comment:
When("called with general error", func() {
			    It("should return an appropriate error", func() {
```suggestion
		When("called with general error", func() {
			It("should return an appropriate error", func() {
```